### PR TITLE
Add documentation for recording rule backfiller

### PR DESCRIPTION
**Pull request by JessicaGreben**
This PR adds documentation for Issue #11 and #8667.

### CODE REVIEW

File: cmd/promtool/main.go [hunk 1]
```diff
@@ -148,11 +148,11 @@ func main() {
 	// TODO(aSquare14): add flag to set default block duration
 	importFilePath := openMetricsImportCmd.Arg("input file", "OpenMetrics file to read samples from.").Required().String()
 	importDBPath := openMetricsImportCmd.Arg("output directory", "Output directory for generated blocks.").Default(defaultDBPath).String()
-	importRulesCmd := importCmd.Command("rules", "Create new blocks of data from Prometheus data for new rules from recording rule files.")
+	importRulesCmd := importCmd.Command("rules", "Create blocks of past data for new recording rules.")
```
Comment (roidelapluie) [inline cmd/promtool/main.go:151]:
```suggestion
	importRulesCmd := importCmd.Command("rules", "Create blocks of data for new recording rules.")
```

We do not enforce the data to be from the paste.

File: docs/storage.md [hunk 2]
```diff
@@ -150,3 +150,50 @@ promtool tsdb create-blocks-from openmetrics <input file> [<output directory>]
 ```
 
 After the creation of the blocks, move it to the data directory of Prometheus. If there is an overlap with the existing blocks in Prometheus, the flag `--storage.tsdb.allow-overlapping-blocks` needs to be set. Note that any backfilled data is subject to the retention configured for your Prometheus server (by time or size).
+
+## Backfilling for Recording Rules
+
+### Overview
+
+When a new recording rule is created there is no historical data for it, rather recording rule data only exists from the creation time on. However there is a `promtool` subcommand that makes it possible to create historical recording rule data.
```
Comment (roidelapluie) [inline docs/storage.md:158]:
```suggestion
When a new recording rule is created, there is no historical data for it, Recording rule data only exists from the creation time on. `promtool` makes it possible to create historical recording rule data.
```

File: docs/storage.md [hunk 3]
```diff
@@ -150,3 +150,50 @@ promtool tsdb create-blocks-from openmetrics <input file> [<output directory>]
...
+
+### Usage
+
+```
```
Comment (roidelapluie) [inline docs/storage.md:162]: This copy and paste seems fragile, I would remove it.

Comment (roidelapluie) [reply docs/storage.md:162]: Maybe say: To see all the options, use `promtool tsdb create-blocks-from rules --help`

File: docs/storage.md [hunk 4]
```diff
@@ -150,3 +150,50 @@ promtool tsdb create-blocks-from openmetrics <input file> [<output directory>]
...
+
+Example usage:
+```
+$ promtool tsdb create-blocks-from rules \
+    --start 1617079873 \
+    --end 1617097873 \
+    --url http://mypromserver.com:9090 \
+    rules.yaml,rules2.yaml
```
Comment (roidelapluie) [inline docs/storage.md:188]:
```suggestion
    rules.yaml rules2.yaml
```
I think this is space separated (or at least it should be)

File: docs/storage.md [hunk 5]
```diff
@@ -150,3 +150,50 @@ promtool tsdb create-blocks-from openmetrics <input file> [<output directory>]
...
+```
+
+The recording rule files provided should comply with Prometheus rule documentation [here](https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/).
```
Comment (roidelapluie) [inline docs/storage.md:191]:
```suggestion
The recording rule files provided should be a normal [Prometheus rules file](https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/).
```

File: docs/storage.md [hunk 6]
```diff
@@ -150,3 +150,50 @@ promtool tsdb create-blocks-from openmetrics <input file> [<output directory>]
...
+
+The output of `promtool tsdb create-blocks-from rules` command is a directory that contains blocks with the historical rule data for all rules in the recording rule files. By default the output directory is `data/`. In order to make use of this new block data, the blocks must be moved to a running Prometheus instance that has the flag `--storage.tsdb.allow-overlapping-blocks` enabled. Once moved, the new blocks will merge with existing blocks when the next compaction runs.
+
+A couple of important things to note:
+- If you run the rule backfiller multiple times with the overlapping start/end times, blocks containing the same data will be created each time the rule backfiller is run.
+- Output blocks are in 2 hr chunks.
```
Comment (roidelapluie) [inline docs/storage.md:197]: I would not specify that as we plan to have this configurable.

File: docs/storage.md [hunk 7]
```diff
@@ -150,3 +150,50 @@ promtool tsdb create-blocks-from openmetrics <input file> [<output directory>]
...
+- All rules in the recording rule files will be evaluated.
+- If the `interval` is set in the recording rule file that will take priority over the `eval-interval` flag in the rule backfill command.
```
Comment (roidelapluie) [inline docs/storage.md:199]: I would add that as a limitation, rules in the same group can not see the results of the previous rules.
I would also add that we ignore alerts.

File: docs/storage.md [hunk 8]
```diff
@@ -150,3 +150,50 @@ promtool tsdb create-blocks-from openmetrics <input file> [<output directory>]
...
+    --end 1617097873 \
+    --url http://mypromserver.com:9090 \
+    rules.yaml,rules2.yaml
+```
+
+The recording rule files provided should comply with Prometheus rule documentation [here](https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/).
+
+The output of `promtool tsdb create-blocks-from rules` command is a directory that contains blocks with the historical rule data for all rules in the recording rule files. By default the output directory is `data/`. In order to make use of this new block data, the blocks must be moved to a running Prometheus instance that has the flag `--storage.tsdb.allow-overlapping-blocks` enabled. Once moved, the new blocks will merge with existing blocks when the next compaction runs.
+
+A couple of important things to note:
```
Comment (roidelapluie) [inline docs/storage.md:195]: Should we have a subtitle for this? like `limitations` ?

**Review by roidelapluie — COMMENTED** [review]
Thank you for this!

File: docs/storage.md (hunk 3 above)
Comment (bwplotka) [reply docs/storage.md:162]: We could setup auto-generated one, but if we do, let's make Pr at some point for all commands (:

**Review by bwplotka — APPROVED** [review]
Thanks a lot! It looks great, but I agree with @roidelapluie suggestions. Let's address them and then LGTM 🚀

File: docs/storage.md (hunk 3 above)
Comment (JessicaGreben) [reply docs/storage.md:162]: @bwplotka Is there somewhere we already have the auto-generated one setup? Or would that be something new? If we want to proceed with that, is it worth creating an issue for it?

File: docs/storage.md (hunk 4 above)
Comment (JessicaGreben) [reply docs/storage.md:188]: good catch.

File: docs/storage.md [hunk 9]
```diff
@@ -150,3 +150,33 @@ promtool tsdb create-blocks-from openmetrics <input file> [<output directory>]
...
+    --end 1617097873 \
+    --url http://mypromserver.com:9090 \
+    rules.yaml rules2.yaml
+```
+
+The recording rule files provided should be a normal [Prometheus rules file](https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/).
+
+The output of `promtool tsdb create-blocks-from rules` command is a directory that contains blocks with the historical rule data for all rules in the recording rule files. By default the output directory is `data/`. In order to make use of this new block data, the blocks must be moved to a running Prometheus instance data dir `storage.tsdb.path` that has the flag `--storage.tsdb.allow-overlapping-blocks` enabled. Once moved, the new blocks will merge with existing blocks when the next compaction runs.
+
+### Limitations
```
Comment (roidelapluie) [inline docs/storage.md:177]: Can we have an extra blank line here? please

**Review by roidelapluie — APPROVED** [review]

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**